Anlab.Mvc/Services/IOrderService.cs

```diff
@@ -144,7 +145,18 @@ private TestDetails[] CalculateTestDetails(Order order)
             order.ClientName = order.ClientName;
             order.JsonDetails = orderToCopy.JsonDetails;
             var orderDetails = order.GetOrderDetails();
-            var tests = CalculateTestDetails(order);
+            var tests = CalculateTestDetails(order).ToList();
+
+            var groupTestIds = tests.Where(a => a.Id.StartsWith("G-")).Select(a => a.Id).ToArray();

```

Comment:
@jSylvestre since this is in-memory I think it'd be worth doing an ordinal string insensitive match, since I could totally see them messing up and doing a lowercase G.  You should just be able to add the comparer as an overload to the startsWith